tests/compliance/test_iam_compliance.py

```diff
@@ -0,0 +1,148 @@
+            "last_used": datetime.now(),
+        },
+    ]
+    keys_account2 = [
+        {
+            "id": "key3_id",
+            "user_name": "test_user3_good",
+            "created": datetime.now() - timedelta(days=1),
+        },
+        {
+            "id": "key4_id",
+            "user_name": "test_user4_old",
+            "created": datetime.now() - timedelta(days=999),
+            "last_used": datetime.now() - timedelta(days=1),
+        },
+    ]
+    account1 = create_account()
+    account2 = create_account()
+    audit = create_audit(
+        [(create_account_report(account1, keys_account1)), (create_account_report(account2, keys_account2))]
+    )
+
+    notifications = IamCompliance().analyse(audit)
+
+    assert sorted(notifications, key=lambda x: x.item) == [
+        Findings(
+            account=account1,
+            compliance_item_type="iam_access_key",
+            item="key1_id",
+            findings={EXPECTED_OLD_KEY_VIOLATION},
+            description="this key is 31 days old and belongs to test_user1_old",
+        ),
+        Findings(
+            account=account1,
+            compliance_item_type="iam_access_key",
+            item="key2_id",
+            findings={EXPECTED_OLD_KEY_VIOLATION},
+            description="this key is 100 days old, belongs to test_user2_old and was last used today",
+        ),
+        Findings(
+            account=account2,
+            compliance_item_type="iam_access_key",
+            item="key3_id",
+            findings=set(),
+            description="this key is 1 day old and belongs to test_user3_good",
+        ),
+        Findings(
+            account=account2,
+            compliance_item_type="iam_access_key",
+            item="key4_id",
+            findings={EXPECTED_OLD_KEY_VIOLATION},
+            description="this key is 999 days old, belongs to test_user4_old and was last used yesterday",
+        ),
+    ]
+
+
+def create_account_report(account: Account, access_keys: List[Dict[str, Any]]):
+    dates_as_strings = []
+    for access_key in access_keys:
+        key = {
+            "id": access_key["id"],
+            "user_name": access_key["user_name"],
+            "created": access_key["created"].isoformat(),
+        }
+        if "last_used" in access_key:
+            key["last_used"] = access_key["last_used"].isoformat()
+
+        dates_as_strings.append(key)
+    return {
+        "account": {
+            "identifier": account.identifier,
+            "name": account.name,
+        },
+        "description": "audit_iam",
+        "results": {"iam_access_keys": dates_as_strings},
+    }
+
+
+def create_account() -> Account:
+    return Account(
+        identifier=str(random.randrange(100000000000, 1000000000000)),
+        name=f"account-{''.join(random.choices(string.ascii_letters, k=15))}",
+    )
+
+
+def create_audit(report: List[Dict[str, Any]]) -> Audit:
+    return Audit(type="iam_access_keys", report=report)
```

Comment:
Might be worth moving this over to https://github.com/hmrc/platsec-compliance-alerting/blob/main/tests/test_types_generator.py as this would likely be useful to other compliance tests.